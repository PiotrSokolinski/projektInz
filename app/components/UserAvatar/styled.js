import styled from 'styled-components'

export const Container = styled.div`
  align-items: center;
  background-color: transparent;
  background-position: center;
  background-repeat: no-repeat;
  border-radius: 100vw;
  display: flex;
  justify-content: center;
  position: relative;

  .avatar-image {
    border-radius: 100vw;
    object-fit: cover;
  }

  &.tiny .avatar-image {
    height: 20px;
    width: 20px;
  }

  &.small .avatar-image {
    height: 30px;
    width: 30px;
  }

  &.middle .avatar-image {
    height: 60px;
    width: 60px;
  }

  &.big .avatar-image {
    height: 90px;
    width: 90px;
  }

  &.large .avatar-image {
    height: 180px;
    width: 180px;
  }
`
