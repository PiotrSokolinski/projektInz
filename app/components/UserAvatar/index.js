/**
 *
 * UserAvatar
 *
 */

import React from 'react'
import PropTypes from 'prop-types'

import { Icons } from 'themes'

import { Container } from './styled'

const UserAvatar = ({ image, size }) => {
  const imageHasUrl = image !== '' && image !== null && image !== undefined
  return (
    <Container className={size}>
      <img className="avatar-image" src={(imageHasUrl && image) || Icons.emptyAvatar} alt="avatar" />
    </Container>
  )
}

UserAvatar.propTypes = {
  image: PropTypes.any,
  size: PropTypes.oneOf(['tiny', 'small', 'middle', 'big', 'large']),
}

UserAvatar.defaultProps = {
  image: Icons.emptyAvatar,
  size: 'small',
}

export default UserAvatar
