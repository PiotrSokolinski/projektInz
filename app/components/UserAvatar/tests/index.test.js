/**
 *
 * Tests for UserAvatar
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'

import UserAvatar from '../index'

describe('<UserAvatar />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(<UserAvatar />)
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<UserAvatar />)
    expect(firstChild).toMatchSnapshot()
  })
})
