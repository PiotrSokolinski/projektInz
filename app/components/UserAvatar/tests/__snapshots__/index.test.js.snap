// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAvatar /> Should render and match the snapshot 1`] = `
<div
  class="small styled__Container-vinm4t-0 gDIJTt"
>
  <img
    alt="avatar"
    class="avatar-image"
    src="IMAGE_MOCK"
  />
</div>
`;
